handbook/unity.adoc

```diff
@@ -0,0 +1,158 @@
+
+[source]
+----
+container.add 'udp', new org.arl.unet.link.UdpLink()              //<1>
+container.add 'wormhole', new org.arl.unet.wormhole.Wormhole()    //<2>
+wormhole.dsp = 'udp'                                              //<3>
+----
+
+<1> Adds a UdpLink(<<Wired and over-the-air links>>) agent that implements a link protocol over UDP/IP for use over wired/wireless IP networks. 
+<2> Adds a Wormhole agent which allows the fjåge messages to be sent between containers over a Unet link. 
+<3> Tells the Wormhole agent to use the UdpLink as the Unet link to share information.
+
+Execute these three lines of code on both the assiting and main receiver (i.e., Node B and Node C).
+
+*Additional set up on the assisting receiver*:
+
+From the assisting receiver, we may not want to share the information from all agents, instead, we are interested in messages that are published only on the `Physical` agent’s topic. Therefore, we need an additional setting on the assiting receiver to forward only the messages received on the `Physical` agent’s topic to be sent over the Wormhole link.
+
+[source]
+----
+wormhole.publish = [topic(phy), topic(phy, Physical.SNOOP)]       //<1>
+----
+
+<1> Tells the wormhole agent to only forward the messages received on `Physical` agent's topic.
+
+Run the above one line of code on the assiting receiver (Node B).
+
+That’s all there is to set up the receiver nodes to share the information and cooperate (in this case over a `UdpLink`). With this set up any message received on the `Physical` agent’s topic on the assisting receivers will also be available at the main receiver.
+
+==== Add the Unity agent on the main receiver
+
+If you happen to have access to the `Unity` agent, it will be available by default in the stack and you don't have to add this agent to the container. However, if you do not have this added in the stack by default, this is how you add it on the main receiver (Node C).
+
+[source]
+----
+container.add 'unity', new org.arl.unet.diversity.Unity()         //<1>
+----
+
+<1> Adds the `Unity` agent to the stack.
+
+Now, you can take a look at the parameters of the `Unity` agent, type unity on the node’s web shell interface: 
+
+[source]
+----
+> unity
+« Spatial diversity agent »
+
+Combines copies of information received from multiple
+receivers to decode packets.
+
+[org.arl.unet.unity.UnityParam]
+  assisters = []
+  enable = false
+  maxAssisterRange = 100.0
+  phy = phy
+----
+
+The `assisters` parameter is used to store the `Physical` agent names running on the assisting receivers that are assisting the main receiver. The user might also want to change the default value of `maxAssisterRange` if the furthest assisting receiver to the main receiver is at a distance larger than 100 m.
```

Comment:
Since main receiver can actually talk to the assisting receiver, is it possible for them to do ranging and update the `maxAssisterRange` automatically? For future revisions of Unity during agent loading?

Reply:
That sounds like a good idea for the future updates on this agent.